### PR TITLE
Add entity api doc

### PR DESCRIPTION
**Pull request by mooreds**
This is the entity api doc.

A couple of issues:

* what is email in this context? https://github.com/FusionAuth/fusionauth-api/blob/master/src/main/java/io/fusionauth/api/service/search/DatabaseEntitySearchEngine.java#L59 
* @jobannon is adding some search stuff to the core concepts. I marked where we needed to update the links with an `XXX`.

### CODE REVIEW

File: site/docs/v1/tech/apis/_entity-grant-response-body-base.adoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
+[api]
+ifeval::["{base_field_name}" != "entityGrant"]
+[field]#{base_field_name}# [type]#[Array]#::
+The list of Entity Grant objects.
+endif::[]
+
+[field]#{base_field_name}.data# [type]#[Object]#::
+An object that can hold any information about the Entity Grant that should be persisted. 
+
```
Comment (bguyza) [inline site/docs/v1/tech/apis/_entity-grant-response-body-base.adoc:9]:
```suggestion

[field]#{base_field_name}.entity# [type]#[Entity]#::
The entity to which access is granted.
```

**Review by bguyza — CHANGES_REQUESTED** [review]
First pass complete.  Nice job!

There is a lot going on in this PR so I'll be happy to do another pass once you've had a chance to address comments here.  I tried to be as exhaustive as possible.  Let me know if you have any questions.

File: site/docs/src/json/entities/response.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "entity": {
+    "clientId": "092dbded-30af-4149-9c61-b578f2c72f59",
+    "clientSecret": "+fcXet9Iu2kQi61yWD9Tu4ReZ113P6yEAkr32v6WKOQ=",
+    "data": {
+      "companyType": "Legal"
+    },
+    "entityType": {
+      "id": "092dbded-30af-4149-9c61-b578f2c72f57",
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/src/json/entities/response.json:9]: It looks like the current FusionAuth behavior is to only respond with the id, so let's follow that.

[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/src/json/entities/response.json:9]: Actually, I'm wrong, it does include the type info. will add that.

File: site/docs/v1/tech/apis/_entity-response-body-base.adoc [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+[field]#{base_field_name}.lastUpdateInstant# [type]#[Long]#::
+The link:/docs/v1/tech/reference/data-types/#instants[instant] that the Entity was last updated in the FusionAuth database.
+
+[field]#{base_field_name}.name# [type]#[String]#::
+A descriptive name for the Entity (i.e. "Raviga" or "Email Service").
+
+[field]#{base_field_name}.parentId# [type]#[UUID]#::
+The parent Entity Id. Parents are consulted for permission checks and allow grouping of related Entities.
+
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/v1/tech/apis/_entity-response-body-base.adoc:30]: So it looks like parent ids are in the schema but don't have any supporting code yet, so I'll remove it.

**Comment by mooreds** [comment]
So I ended up breaking out grants and moving all the apis down under `entity management`. 

Incorporated all your suggestions and wrote scripts against the api.

**Review by bguyza — APPROVED** [review]
This LGTM.  I really like how you reorganized the content as well.

**Pull request merged**